### PR TITLE
REF: `axis in [..]` with `_get_axis_number()` 

### LINKED ISSUE #43048: REF: use `_get_axis_number` instead of `if axis in [...]`:
This PR regards searching the **non-test codebase** (i.e. don't find code examples in pandas/tests) for examples like the following:

Try to find and replace logic like:

```python
if axis in [0, "index"]:
    ...
elif axis in [1, "columns"]:
    ...
else:
    ...
```

with the standard codebase method `get_axis_number(axis)`. This is a DataFrame method so often you need use local DataFrame objects to call this method.

It may also be necessary to change `axis` args in internal methods to consistently accept the same input format compatible with this method, i.e. axis as an integer (0 or 1).

This is an open issue and contributions are welcome to solve individual files in the code. As an example see #43078 or other merged contributions below.

- [x] `style.py` #43078
- [x] `test_label_or_level_utils.py` #43131
- [x] #43121

### PR DESCRIPTION
**Pull request by HardevKhandhar**
- [x] xref #43048 
- [x] Refactored test_frame_apply.py 
- [x] Created if-else constructs by replacing `if axis in [...]` by `_get_axis_number`

### CODE REVIEW

**Review by attack68 — CHANGES_REQUESTED** [review]
lgtm pending green tests. small comment:

you can convert some of the 4-liners into 1 lines by replacing the pattern:

```python 
if axis == 0:
    var = value
else:
    var = other_value
```
 
with `var = value if axis == 0 else other_value`.

If it fits on a single line you can even write:

```python
var = value if frame._get_axis_number(axis) == 0 else other_value
```

**Comment by HardevKhandhar** [comment]
Done. Please check.
Thank you.

**Comment by attack68** [comment]
lgtm pending green

**Review by attack68 — APPROVED** [review]

File: pandas/tests/apply/test_frame_apply.py [hunk 1]
```diff
@@ -995,7 +995,8 @@ def test_consistency_for_boxed(box, int_frame_const_col):
 
 
 def test_agg_transform(axis, float_frame):
-    other_axis = 1 if axis in {0, "index"} else 0
+    axis = float_frame._get_axis_number(axis)
+    other_axis = 1 if axis == 0 else 0
```
Comment (rhshadrach) [inline pandas/tests/apply/test_frame_apply.py:999]: This makes it so we aren't testing cases (e.g. passing "index" to apply). I don't think we want to do this in the tests, only in the pandas code itself.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/tests/apply/test_frame_apply.py (hunk 1 above)
Comment (HardevKhandhar) [reply pandas/tests/apply/test_frame_apply.py]: Changes reverted to original for `test_agg_function` 
Thank you.

Comment (rhshadrach) [reply pandas/tests/apply/test_frame_apply.py]: Thanks @HardevKhandhar. The same remark applies to your change to the other test here. While the changes you made were certainly of the right form, we just don't want to apply them to the tests. I recommend searching the code in the rest of pandas (outside of tests) and find examples to make this change to.

Comment (HardevKhandhar) [reply pandas/tests/apply/test_frame_apply.py]: Alright, so, I need to revert back to the changes in all the test functions right?

Comment (rhshadrach) [reply pandas/tests/apply/test_frame_apply.py]: If we're making no changes, then I think we can close this PR. But just to be sure - I think you're going in the right direction here, just need to make the modification to cases that occur outside of `pandas.tests`. Once you find some, make the changes similar to what you did here and submit a new PR.

Comment (HardevKhandhar) [reply pandas/tests/apply/test_frame_apply.py]: Sure, I am closing the PR.
Thanks a lot @rhshadrach :)

**Pull request closed without merging**